### ISSUE TITLE
Feature Request: Extensions/plug-ins

### ISSUE THREAD

**Issue by munsonbh** [heyman/heynote#75]
If you make this extensible, you won’t have to build the long tail of micro-features yourself!

**Comment by heyman** [comment]
Yes, that would be great to have ultimately. Having a plugin/extension system comes with a high cost, though. You have to come up with a good API, and once you release it, you've committed to keeping it backwards compatible (preferably indefinitely). There are still features that I think Heynote itself should have that I think would make sense to implement before creating an extension API.
